.github/workflows/CI-build-test-publish.yml

```diff
@@ -0,0 +1,149 @@
+        required: true
+      GH_TOKEN:
+        required: true
+      QA_SLACK_WEBHOOK_URL:
+        required: true
+
+env:
+  DOCKER_HUB_ORG: horizenlabs
+  DOCKER_IMAGE_BUILD_NAME: zkverify
+
+jobs:
+  build:
+    runs-on:
+      group: zkVerify
+    name: Build Docker image
+    outputs:
+      artifact_name: ${{ steps.create-tar-file.outputs.artifact_name }}
+  
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.git_ref }}
+          fetch-depth: 0
+
+      - name: Build docker image
+        run: |
+          if [[ "${{ inputs.chain }}" == "solo" ]]; then
+            docker build --build-arg PROFILE=production -f "docker/dockerfiles/zkv-node.Dockerfile" -t "${DOCKER_HUB_ORG}/zkverify-solo" .
+            echo "DOCKER_IMAGE_NAME=${DOCKER_HUB_ORG}/zkverify-solo" >> $GITHUB_ENV
+          elif [[ "${{ inputs.chain }}" == "relay" ]]; then
+            docker build --build-arg PROFILE=production -f "docker/dockerfiles/zkv-relay.Dockerfile" -t "${DOCKER_HUB_ORG}/zkverify-relay" .
+            echo "DOCKER_IMAGE_NAME=${DOCKER_HUB_ORG}/zkverify-relay" >> $GITHUB_ENV
+          fi
+      
+      - name: Save the Docker image as a tarball
+        id: create-tar-file
+        run: |       
+          ARTIFACT_NAME="zkverify-${{ inputs.chain }}"
+          TAR_FILE="${{github.workspace}}/${ARTIFACT_NAME}.tar"
+
+          echo "artifact_name=$ARTIFACT_NAME" >> $GITHUB_OUTPUT
+          echo "TAR_FILE=$TAR_FILE" >> $GITHUB_ENV
+
+          docker save "${DOCKER_IMAGE_NAME}" -o "${TAR_FILE}"
+
+      - name: Upload Docker Image Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.create-tar-file.outputs.artifact_name }}
+          path: ${{ env.TAR_FILE }}
+          if-no-files-found: error
+          retention-days: 1
+          compression-level: 0
+          overwrite: true
+  
+  execute-e2e-tests:
+    runs-on: ubuntu-latest
+    needs: build
+    name: Execute E2E Tests
+    outputs:
+      test_status: ${{ steps.run-test.outputs.test_status }} 
+    steps:
+      - name: E2E
+        run: echo "execute-e2e-tests with ${{ needs.build.outputs.artifact_name }}"  
+
+      - uses: actions/checkout@v4
+        with:
+          repository: HorizenLabs/zkVerify-qa
+          ref: main
+          clean: true
+          fetch-depth: 1
+
+      - name: Download zkVerify Docker image artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.build.outputs.artifact_name }}
+          path: ./
+
+      - name: Load Docker image and rename
+        run: |
+          IMAGE_NAME="$(docker load -i "${GITHUB_WORKSPACE}/${{ needs.build.outputs.artifact_name }}.tar" | awk '/Loaded image:/ { print $3 }')"
+          echo $(docker images -q "${IMAGE_NAME}" > /dev/null 2>&1 && echo "Image exists..." || echo "Image missing...")
+          echo "Loaded image ${IMAGE_NAME}"
+          TEST_TAG="test"
+          echo "TEST_TAG=${TEST_TAG}" >> $GITHUB_ENV
```

Comment:
Why not to use `test` explicitly without the VAR if we are not setting it dynamically or changing via ENV vars :thinking: